nexus/src/app/rack.rs

```diff
@@ -737,7 +738,10 @@ impl super::Nexus {
         };
 
         debug!(self.log, "Listing sleds");
-        let sleds = self.db_datastore.sled_list(opctx, &pagparams).await?;
+        let sleds = self
+            .db_datastore
+            .sled_list(opctx, &pagparams, SledFilter::All)
```

Comment:
Should this be `SledFilter::InService`? I think with `::All` we wouldn't allow re-initializing a previously-expunged sled (because we'd still see its baseboard here)

Reply:
I wasn't sure about this, but I'm pretty sure you are right. Good catch!

Reply:
Fixed in 8b07fa885bdf94ed2009e1bda4da5d34dd6b4c72